src/libs/ReportUtils.ts

```diff
@@ -3459,14 +3472,69 @@ function shouldHideReport(report: OnyxEntry<Report>, currentReportId: string): b
     return parentReport?.reportID !== report?.reportID && !isChildReportHasComment;
 }
 
+function transactionHasViolation(transactionID: string, transactionViolations: TransactionViolations): boolean {
+    const violations: TransactionViolation[] = transactionViolations[transactionID];
+    return violations.some((violation: TransactionViolation) => violation.type === 'violation');
+}
+
+/**
+ *  Checks to see if a report's parentAction is a money request that contains a violation
+ */
+
+function transactionThreadHasViolations(
+    report: Report,
+    transactionViolations: TransactionViolations,
+    reportActions?: OnyxCollection<ReportActions> | null,
+    parentReportAction?: ReportAction | null,
```

Comment:
Let's make reportActions mandatory and remove parentReportAction

Reply:
done.